packages/block-library/src/quote/transforms.js

```diff
@@ -74,44 +153,47 @@ const transforms = {
 				return (
 					node.nodeName === 'BLOCKQUOTE' &&
 					// The quote block can only handle multiline paragraph
-					// content with an optional cite child.
-					Array.from( node.childNodes ).every(
-						isParagraphOrSingleCite
-					)
+					// content with an optional figcaption child.
+					Array.from( node.childNodes ).every( isAllowedNode )
 				);
 			},
-			schema: ( { phrasingContentSchema } ) => ( {
-				blockquote: {
-					children: {
-						p: {
-							children: phrasingContentSchema,
-						},
-						cite: {
-							children: phrasingContentSchema,
-						},
-					},
-				},
-			} ),
+			transform: ( node ) =>
+				createBlock(
+					'core/quote',
+					{},
+					rawHandler( {
+						HTML: node.innerHTML,
+						mode: 'BLOCKS',
+					} )
+				),
 		},
 	],
 	to: [
+		{
+			type: 'block',
+			blocks: [ 'core/group' ],
+			transform: ( { attribution }, innerBlocks ) =>
+				createBlock(
+					'core/group',
+					{},
+					attribution
+						? [
+								...innerBlocks,
+								createBlock( 'core/paragraph', {
+									content: attribution,
+								} ),
+						  ]
+						: innerBlocks
+				),
+		},
```

Comment:
I'm not sure if this transform provides value. If you really want to switch from a quote to a group, you can unquote it and then group.